### ISSUE TITLE
(I am willing to make a PR) Make RenderEditable and TextPainter's `text` field not-null

### ISSUE THREAD

**Issue by fzyzcjy** [flutter/flutter#80207]
Hi thanks for the framework! When digging into the source code, I find a place where the nullability can be enhanced: The `RenderEditable.text` and `TextPainter.text`. Originally, those were nullable. But I have tried to change them to not-null (with corresponding changes in other locations), and no compiler errors occur. Moreover, by doing this, we can **remove 23 `!`(nullability) checks** (which will, otherwise, be error-prone and make flutter run slower).

I am willing to make a PR!

Details of what I have changed:

```
  TextSpan? get text => _textPainter.text as TextSpan?;
  final MyTextPainter _textPainter;
  set text(TextSpan? value) {

and

  InlineSpan? get text => _text;
  InlineSpan? _text;
  set text(InlineSpan? value) {
```

to:

```
  TextSpan get text => _textPainter.text as TextSpan;
  final MyTextPainter _textPainter;
  set text(TextSpan value) {

and

  InlineSpan get text => _text;
  InlineSpan _text;
  set text(InlineSpan value) {
```

Of course, with some other modifications (e.g. make the constructor accept not-null arguments). It works pretty well without any compilation errors!

**Comment by LongCatIsLooong** [comment]
Makes sense to me as the `TextPainter` can't do anything with its text being `null`. But `TextPainter.text` being nullable may be intentional, as I'd expect people to use `TextPainter` builder-style:
```dart
final TextPainter painter = TextPainter()
  ..text = ...
```
But in any case, we should be able to mark the `text` attribute `late`. Feel free to open a PR for the change,  I'm happy to run customer tests on the PR and see if it is a breaking change.

**Comment by amanv8060** [comment]
@LongCatIsLooong so when using builder pattern people won't be able to enter null specifically? 

If yes, found two places where the framework itself sets it to null, so there must be more people doing so.

**Comment by fzyzcjy** [comment]
@amanv8060 Agree
